### PR TITLE
Update dates to handle timezone offset

### PR DESCRIPTION
**Pull request by germsvel**
Update date decoding to handle timezone offset. We take in the offset in minutes as a config option into the Elm app and pass it into the decoder so that we handle the offset when the dates are coming into our application.

### CODE REVIEW

File: src/BookingWidget/Api.elm [hunk 1]
```diff
@@ -20,9 +22,12 @@ import Http exposing (Request)
 import Json.Decode exposing (Decoder)
 
 
-queryAvailability : ApiCredentials -> ProductId -> Date -> Cmd Msg
-queryAvailability apiCreds productId date =
+queryAvailability : Model -> ProductId -> Date -> Cmd Msg
+queryAvailability model productId date =
     let
+        apiCreds =
+            model.apiCredentials
```
Comment (JoelQ) [inline src/BookingWidget/Api.elm]: I'm curious why you needed this assignment?

File: src/BookingWidget/Api.elm [hunk 2]
```diff
@@ -40,8 +45,11 @@ queryAvailability apiCreds productId date =
 
         handler =
             Http.send handleAvailabilityResponse
+
+        offsetInMinutes =
+            model.offsetInMinutes
```
Comment (JoelQ) [inline src/BookingWidget/Api.elm]: Thoughts on calling the value directly inline? Alternatively, thoughts on destructuring the record in the signature?

File: src/BookingWidget/Api/Decoders.elm [hunk 3]
```diff
@@ -15,24 +15,41 @@ import Json.Decode exposing (bool, int, list, maybe, string, Decoder)
...
+availabilityDecoder : OffsetInMinutes -> Decoder DateAvailability
+availabilityDecoder offsetInMinutes =
     decode DateAvailability
-        |> optional "date" (Json.Decode.map (Date.fromString >> Result.toMaybe) string) Nothing
+        |> optional "date" (Json.Decode.map (formatDate offsetInMinutes) string) Nothing
         |> optional "soldOut" (Json.Decode.map soldOutBoolToAvailability bool |> maybe) Nothing
 
 
+formatDate : OffsetInMinutes -> String -> Maybe Date.Date
+formatDate offsetInMinutes =
```
Comment (JoelQ) [inline src/BookingWidget/Api/Decoders.elm]: This function is parsing the date, not formatting it right?

File: src/BookingWidget/Api/Decoders.elm [hunk 4]
```diff
@@ -15,24 +15,41 @@ import Json.Decode exposing (bool, int, list, maybe, string, Decoder)
...
+handleTimeZone : OffsetInMinutes -> Date.Date -> Date.Date
+handleTimeZone (OffsetInMinutes offsetInMinutes) =
+    Date.fromTime
+        << ((+) (minutesToMilliseconds offsetInMinutes))
+        << Date.toTime
+
+
+minutesToMilliseconds : Float -> Float
+minutesToMilliseconds v =
+    v * 60 * 1000
```
Comment (JoelQ) [inline src/BookingWidget/Api/Decoders.elm]: Thoughts on using the helpers provided by the core `Time` module for this?

```elm
minutesToMilliseconds : Float -> Float
minutesToMilliseconds =
  Time.inMilliseconds << Time.minutes
```

File: src/BookingWidget/Api.elm (hunk 1 above)
Comment (germsvel) [inline src/BookingWidget/Api.elm]: Didn't need it. I think it came along as part of the refactoring. But it does seem unnecessary. I'll make a change.

File: src/BookingWidget/Api.elm (hunk 2 above)
Comment (germsvel) [inline src/BookingWidget/Api.elm]: Yep. Both good ideas. I'll see what fits well and update.

File: src/BookingWidget/Api/Decoders.elm (hunk 3 above)
Comment (germsvel) [inline src/BookingWidget/Api/Decoders.elm]: Yeah. Maybe got hasty with my naming. Parser is more appropriate.

File: src/BookingWidget/Api/Decoders.elm (hunk 4 above)
Comment (germsvel) [inline src/BookingWidget/Api/Decoders.elm]: I didn't know those existed. I like that!

**Pull request merged**